lib/rubocop/server/cache.rb

```diff
@@ -69,10 +69,12 @@ def cache_root_dir_from_config
             config_yaml = if Gem::Version.new(Psych::VERSION) >= Gem::Version.new('4.0.0')
                             YAML.safe_load_file(config_path, permitted_classes: [Regexp, Symbol])
                           else
-                            YAML.load_file(config_path)
+                            config = YAML.load_file(config_path)
+
+                            config == false ? nil : config
```

Comment:
FYI, this is to avoid the following incompatibility.

```console
% ruby -ryaml -ve "p YAML.load('')"
ruby 3.0.4p208 (2022-04-12 revision 3fa771dded) [x86_64-darwin19]
false

% ruby -ryaml -ve "p YAML.load('')"
ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [x86_64-darwin19]
nil
```

The added test will not pass the CI matrix if this statement doesn't exist.